config/v1/types_apiserver.go

```diff
@@ -51,6 +51,37 @@ type APIServerSpec struct {
 	// is VersionTLS12.
 	// +optional
 	TLSSecurityProfile *TLSSecurityProfile `json:"tlsSecurityProfile,omitempty"`
+	// audit specifies the settings for audit configuration to be applied to all OpenShift-provided
+	// API servers in the cluster.
+	Audit Audit `json:"audit,omitempty"`
+}
+
+// AuditProfileType defines the audit policy profile type.
+type AuditProfileType string
+
+const (
+	// "Default" is the existing default audit configuration policy.
+	AuditProfileDefaultType AuditProfileType = "Default"
+
+	// "WriteRequestBodies" is similar to Default but it logs request and response
+	// HTTP payloads for write requests (create, update, patch)
+	WriteRequestBodiesAuditProfileType AuditProfileType = "WriteRequestBodies"
+
+	// "AllRequestBodies" is similar to WriteRequestBodies, but also logs request
+	// and response HTTP payloads for read requests (get, list).
+	AllRequestBodiesAuditProfileType AuditProfileType = "AllRequestBodies"
+)
+
+type Audit struct {
+	// profile specifies the name of the desired audit policy configuration to be deployed to
+	// all OpenShift-provided API servers in the cluster
+	//
+	// We provide the following profiles
+	// - Default
+	// - WriteRequestBodies
+	// - AllRequestBodies
+	// If unset, the 'Default' profile is used as the default.
+	Profile AuditProfileType `json:"profile,omitempty"`
```

Comment:
add kubebuilder marker for a default `"Default"`

Reply:
plus an enum marker for the three values

Reply:
https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
> plus an enum marker for the three values

still missing

Reply:
@sttts I have put it where the enum is declared - https://github.com/openshift/api/pull/681/files#diff-2a98fd5f6a0f42f75cdc7d0bfbeb4333R62-R63


you want me to move it here?

Reply:
Nope. You way is even better :+1: